### ISSUE TITLE
LLVM 10 support

### ISSUE THREAD

**Issue by margnus1** [nidhugg/nidhugg#74]
We'd like to support compiling against LLVM 10.

Currently, the compile fails with the following error:
```
  CXX      libnidhugg_a-Transform.o
../../src/Transform.cpp: In function ‘bool Transform::transform(llvm::Module&, const Configuration&)’:
../../src/Transform.cpp:60:11: error: ‘initializeCore’ is not a member of ‘llvm’
   60 |     llvm::initializeCore(Registry);
      |           ^~~~~~~~~~~~~~
../../src/Transform.cpp:61:11: error: ‘initializeScalarOpts’ is not a member of ‘llvm’
   61 |     llvm::initializeScalarOpts(Registry);
      |           ^~~~~~~~~~~~~~~~~~~~
../../src/Transform.cpp:62:11: error: ‘initializeObjCARCOpts’ is not a member of ‘llvm’
   62 |     llvm::initializeObjCARCOpts(Registry);
      |           ^~~~~~~~~~~~~~~~~~~~~
../../src/Transform.cpp:63:11: error: ‘initializeVectorization’ is not a member of ‘llvm’
   63 |     llvm::initializeVectorization(Registry);
      |           ^~~~~~~~~~~~~~~~~~~~~~~
../../src/Transform.cpp:64:11: error: ‘initializeIPO’ is not a member of ‘llvm’
   64 |     llvm::initializeIPO(Registry);
      |           ^~~~~~~~~~~~~
../../src/Transform.cpp:65:11: error: ‘initializeAnalysis’ is not a member of ‘llvm’
   65 |     llvm::initializeAnalysis(Registry);
      |           ^~~~~~~~~~~~~~~~~~
../../src/Transform.cpp:69:11: error: ‘initializeTransformUtils’ is not a member of ‘llvm’
   69 |     llvm::initializeTransformUtils(Registry);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~
../../src/Transform.cpp:70:11: error: ‘initializeInstCombine’ is not a member of ‘llvm’
   70 |     llvm::initializeInstCombine(Registry);
      |           ^~~~~~~~~~~~~~~~~~~~~
../../src/Transform.cpp:71:11: error: ‘initializeInstrumentation’ is not a member of ‘llvm’; did you mean ‘PassInstrumentation’?
   71 |     llvm::initializeInstrumentation(Registry);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~
      |           PassInstrumentation
../../src/Transform.cpp:72:11: error: ‘initializeTarget’ is not a member of ‘llvm’
   72 |     llvm::initializeTarget(Registry);
      |           ^~~~~~~~~~~~~~~~
```

One guess is that LLVM changed its pass manager interface.

**Comment by iamwonseok** [comment]
Hi, margnus1

I've got the same issue. Here is the solution for my case:
#include "llvm/InitializePasses.h" or #include <llvm/InitializePasses.h>
Could you please add the header file in your file then please let me know whether the solution is right or not.

**Comment by margnus1** [comment]
@iamwonseok Thanks! That does seem to be the solution. It should now work on master.

Closed by #75
